code/game/machinery/cryopod.dm

```diff
@@ -41,105 +41,92 @@
+	add_fingerprint(user)
+	ui_interact(user)
 
-	var/dat
+/obj/machinery/computer/cryopod/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "CryopodConsole", name, 400, 480)
+		ui.open()
 
-	if(!( SSticker ))
-		return
+/obj/machinery/computer/cryopod/ui_data(mob/user)
+	var/list/data = list()
+	data["allow_items"] = allow_items
+	data["frozen_crew"] = frozen_crew
+	data["frozen_items"] = list()
 
-	dat += "<hr/><br/><b>[storage_name]</b><br/>"
-	dat += "<i>Welcome, [user.real_name].</i><br/><br/><hr/>"
-	dat += "<a href='?src=[UID()];log=1'>View storage log</a>.<br>"
 	if(allow_items)
-		dat += "<a href='?src=[UID()];view=1'>View objects</a>.<br>"
-		dat += "<a href='?src=[UID()];item=1'>Recover object</a>.<br>"
-		dat += "<a href='?src=[UID()];allitems=1'>Recover all objects</a>.<br>"
+		data["frozen_items"] = frozen_items
 
-	user << browse(dat, "window=cryopod_console")
-	onclose(user, "cryopod_console")
+	var/obj/item/card/id/id_card
+	if(isliving(user))
+		var/mob/living/carbon/human/person = user
+		id_card = person.get_idcard()
+	if(id_card?.registered_name)
+		data["account_name"] = id_card.registered_name
 
-/obj/machinery/computer/cryopod/Topic(href, href_list)
+	return data
+
+/obj/machinery/computer/cryopod/ui_act(action, params)
 	if(..())
-		return 1
+		return
+
+	if(stat & (NOPOWER|BROKEN))
+		return
 
 	var/mob/user = usr
 
 	add_fingerprint(user)
 
-	if(href_list["log"])
-
-		var/dat = "<b>Recently stored [storage_type]</b><br/><hr/><br/>"
-		for(var/person in frozen_crew)
-			dat += "[person]<br/>"
-		dat += "<hr/>"
-
-		user << browse(dat, "window=cryolog")
-
-	if(href_list["view"])
-		if(!allow_items) return
-
-		var/dat = "<b>Recently stored objects</b><br/><hr/><br/>"
-		for(var/obj/item/I in frozen_items)
-			dat += "[I.name]<br/>"
-		dat += "<hr/>"
-
-		user << browse(dat, "window=cryoitems")
-
-	else if(href_list["item"])
-		if(!allowed(user))
-			to_chat(user, "<span class='warning'>Access Denied.</span>")
-			return
-		if(!allow_items) return
-
-		if(frozen_items.len == 0)
-			to_chat(user, "<span class='notice'>There is nothing to recover from storage.</span>")
-			return
+	switch(action)
+		if("one_item")
+			if(!allowed(user))
+				to_chat(user, "<span class='warning'>Access Denied.</span>")
+				return
 
-		var/obj/item/I = input(usr, "Please choose which object to retrieve.","Object recovery",null) as null|anything in frozen_items
-		if(!I)
-			return
+			if(!allow_items)
+				return
 
-		if(!(I in frozen_items))
-			to_chat(user, "<span class='notice'>\The [I] is no longer in storage.</span>")
-			return
+			if(!params["item"])
+				return
 
-		visible_message("<span class='notice'>The console beeps happily as it disgorges [I].</span>")
+			var/obj/item/item = frozen_items[text2num(params["item"])]
```

Comment:
could you perform some sanity/safety check on this parameter before using it to access a list?

just basic stuff like checking to make sure it's a number and that it's not accessing an element out of bounds of the list